packages/tailwindcss/src/utilities.ts

```diff
@@ -1381,70 +1441,74 @@ export function createUtilities(theme: Theme) {
       return `${value}%`
     },
     handle: (value) => [
-      scaleProperties(),
+      ...scaleProperties(),
       decl('--tw-scale-x', value),
       decl('--tw-scale-y', value),
-      decl('scale', 'var(--tw-scale-x) var(--tw-scale-y)'),
     ],
   })
 
-  /**
-   * @css `scale`
-   */
-  functionalUtility('scale-x', {
-    supportsNegative: true,
-    themeKeys: ['--scale'],
-    handleBareValue: ({ value }) => {
-      if (Number.isNaN(Number(value))) return null
-      return `${value}%`
+  suggest('scale', () => [
+    {
+      supportsNegative: true,
+      values: ['0', '50', '75', '90', '95', '100', '105', '110', '125', '150', '200'],
+      valueThemeKeys: ['--scale'],
     },
-    handle: (value) => [
-      scaleProperties(),
-      decl('--tw-scale-x', value),
-      decl('scale', 'var(--tw-scale-x) var(--tw-scale-y)'),
-    ],
-  })
+  ])
+
+  for (let axis of ['x', 'y', 'z']) {
+    /**
+     * @css `scale`
+     */
+    functionalUtility(`scale-${axis}`, {
+      supportsNegative: true,
+      themeKeys: ['--scale'],
+      handleBareValue: ({ value }) => {
+        if (Number.isNaN(Number(value))) return null
+        return `${value}%`
+      },
+      handle: (value) => [...scaleProperties(), decl(`--tw-scale-${axis}`, value)],
+    })
+
+    suggest(`scale-${axis}`, () => [
+      {
+        supportsNegative: true,
+        values: ['0', '50', '75', '90', '95', '100', '105', '110', '125', '150', '200'],
+        valueThemeKeys: ['--scale'],
+      },
+    ])
+  }
 
   /**
    * @css `scale`
    */
-  functionalUtility('scale-y', {
+  functionalUtility('scale-3d', {
```

Comment:
I wonder if there's a way to make this a static utility that acts like a flag instead of a new suite of classes? APIs like `scale-3d-150` feel kinda yuck to me.

If we could do `scale-150 scale-3d` I think that might feel nicer, might have to think through what `scale-x-150 scale-3d` should mean but I think it feels solveable.

Reply:
Yes, that's much better! [Implemented here](https://github.com/tailwindlabs/tailwindcss/pull/13248/commits/03a5c47dd8fa423e6eb4f919c540a38511c276c9).

Effectively, `scale` sets the scale in all three dimensions, but only applies it in two. `scale-3d` applies across all three. `scale-x`, `scale-y`, and `scale-z` can be used to set (or override) the scale in individual dimensions.

In the case of `scale-x-150 scale-3d`, the `scale-3d` has no effect, as the scale in the z-dimension is `1`.

Reply:
`scale-z` also applies in three dimensions, so `scale-3d` is not required alongside `scale-z`.

Reply:
Could we still support a way for 3d transforms to be made in a single utility, like `scale3d-[2,1.5,3]`

Especially for something that gets transitioned as often as transform properties, it would be useful to be able to interpret these values onto element and reference them via a single var.

```html
<div
  style={{
    '--scale3d': scale3d
  }}
  class="scale3d-[--scale3d]"
/>
```

Without this, we could still interpret values this way, but we would have to break each axis for each transformation into its own CSS variable and connected tail and utility.

```html
<div
  style={{
    '--scale-x': scaleX,
    '--scale-y': scaleY,
    '--scale-z': scaleZ,
  }}
  class="scale-x-[--scale-x] scale-y-[--scale-y] scale-z-[--scale-z] scale-3d"
/>
```

I think, as long as the below still works, we can just use more complex 3d situations like this when needed:

```html
<div
  style={{
    '--scale3d': scale3d
  }}
  class="[scale3d:var(--scale3d)]"
/>
```

Reply:
Yes, an escape hatch for arbitrary values is helpful. I've added that so your first example can be written `scale-[2_1.5_3]` and the last can be simplified to `scale-[var(--scale3d)]`.

Reply:
Thanks! I presume `scale-[--scale3d]` would also work (without `var()`)?

Reply:
Yes, it will support whatever Tailwind CSS v4 supports in general. Bare variables lead to a number of parsing ambiguities, so there's a chance support for them could be deprecated/dropped, but no decisions have been made yet.